hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -764,6 +764,26 @@ public void initialize(URI name, Configuration originalConf)
     }
   }
 
+  /**
+   * If classloader isolation is {@code true}
+   * (through {@link Constants#AWS_S3_CLASSLOADER_ISOLATION}) or not
+   * explicitly set, then the classLoader of the input configuration object
+   * will be set to the input classloader, otherwise nothing will happen.
+   * @param conf configuration object.
+   * @param classLoader isolated classLoader.
+   */
+  private void isolateClassloader(Configuration conf, ClassLoader classLoader) {
```

Comment:
move into S3AUtils, make static. I have a dream of S3AFs getting smaller -or at least no larger